File: backend/src/main/java/com/festago/stage/dto/StageResponse.java
package com.festago.stage.dto;

import com.festago.stage.domain.Stage;
import java.time.LocalDateTime;

public record StageResponse(
    Long id,
    Long festivalId,
    LocalDateTime startTime,
    LocalDateTime ticketOpenTime,
    String lineUp) {

    public static StageResponse from(Stage stage) {
        return new StageResponse(stage.getId(), stage.getFestival().getId(), stage.getStartTime(),
            stage.getTicketOpenTime(), stage.getLineUp());
    }
}

File: android/festago/data/src/main/java/com/festago/festago/data/dto/stage/StageResponse.kt
package com.festago.festago.data.dto.stage

import com.festago.festago.data.dto.artist.ArtistResponse
import com.festago.festago.domain.model.stage.Stage
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class StageResponse(
    val id: Long,
    val startDateTime: String,
    val artists: List<ArtistResponse>,
) {
    fun toDomain() = Stage(
        id = id,
        startDateTime = LocalDateTime.parse(startDateTime),
        artists = artists.map { it.toDomain() },
    )
}
